addons/account/models/account_move_send.py

```diff
@@ -1,107 +1,107 @@
-    send_mail_readonly = fields.Boolean(compute='_compute_send_mail_extra_fields')
-    mail_template_id = fields.Many2one(
-        comodel_name='mail.template',
-        string="Email template",
-        domain="[('model', '=', 'account.move')]",
-    )
-    pdf_template_id = fields.Many2one(
-        comodel_name='ir.actions.report',
-        string="Invoice template:",
-        domain="[('is_invoice_report', '=', True)]",
-        default=_get_default_pdf_template_id,
-    )
-    show_pdf_template_menu = fields.Boolean(compute='_compute_show_pdf_template_menu')
-    mail_lang = fields.Char(
-        string="Lang",
-        compute='_compute_mail_lang',
-    )
-    mail_partner_ids = fields.Many2many(
-        comodel_name='res.partner',
-        string="Recipients",
-        compute='_compute_mail_partner_ids',
-        store=True,
-        readonly=False,
-    )
-    mail_subject = fields.Char(
-        string="Subject",
-        compute='_compute_mail_subject_body',
-        store=True,
-        readonly=False,
-    )
-    mail_body = fields.Html(
-        string="Contents",
-        sanitize_style=True,
-        compute='_compute_mail_subject_body',
-        store=True,
-        readonly=False,
-    )
-    mail_attachments_widget = fields.Json(
-        compute='_compute_mail_attachments_widget',
-        store=True,
-        readonly=False,
-    )
-
-    @api.model
-    def default_get(self, fields_list):
-        # EXTENDS 'base'
-        results = super().default_get(fields_list)
-
-        if 'move_ids' in fields_list and 'move_ids' not in results:
-            move_ids = self._context.get('active_ids', [])
-            if any(move.state == 'draft' for move in self.env['account.move'].browse(move_ids)):
-                raise UserError(_("You can't Send & Print invoice(s) in draft state."))
-            results['move_ids'] = [Command.set(move_ids)]
-
-        return results
+    def _get_alerts(self, moves, moves_data):
+        """ Returns a dict of all alerts corresponding to moves with the given context (sending method,
+        edi format to generate, extra_edi to generate).
+        An alert can have some information:
+        - level (danger, info, warning, ...)  (! danger alerts are considered blocking and will be raised)
+        - message to display
+        - action_text for the text to show on the clickable link
+        - action the action to run when the link is clicked
+        """
+        alerts = {}
+        if partners_without_mail := moves.filtered(lambda m: moves_data[m]['sending_method'] == 'email' and not m.partner_id.email).partner_id:
+            alerts['account_missing_email'] = {
+                'level': 'danger' if len(partners_without_mail) == 1 else 'warning',
+                'message': _("Partner(s) should have an email address."),
+                'action_text': _("View Partner(s)"),
+                'action': partners_without_mail._get_records_action(name=_("Check Partner(s) Email(s)")),
```

Comment:
I agree it would be nice to align EDI's in that regard, some use similar checks, which is great because we can easily raise them before even starting the processing.